Ginger/GingerCore/Drivers/Appium/SeleniumAppiumDriver.cs

```diff
@@ -201,78 +224,103 @@ public bool ConnectToAppium()
         {
             try//Adding back the Try-Catch because without it in case of connection issue Ginger crash
             {
-                //Check if we need to use old IP and Port settings
-                if (String.IsNullOrEmpty(AppiumServer)==false) {
-                    //Get IP address and port from old settings - AppiumServer string value
-                    GetPortAndIPFromOldSettings();
+                Uri serverUri = null;
+                if (String.IsNullOrEmpty(AppiumServer) == false)
+                {
+                    try
+                    {
+                        serverUri = new Uri(AppiumServer);
+                    }
+                    catch (Exception)
+                    {
+                        throw new Exception("In-Valid AppiumServer configuration");
+                    }
                 }
+                else
+                {
+                    if (String.IsNullOrEmpty(AppiumServerIP) == true || String.IsNullOrEmpty(AppiumServerPort) == true)
+                    {
+                        throw new Exception("In-Valid AppiumServerIP/AppiumServerPort configuration");
```

Comment:
Codacy found an issue: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6113827)